### ISSUE TITLE
Unable to reproduce `seg_hrnet_w18_small_v1`

### ISSUE THREAD

**Issue by alvinwan** [HRNet/HRNet-Semantic-Segmentation#51]
Thanks for 27488d4, the configuration file is very helpful. With that said, training on 4 GPUs as prescribed, I'm unable to reproduce Cityscapes validation accuracy of 70.3% (attained 65.21%) https://github.com/HRNet/HRNet-Semantic-Segmentation#small-models.

Is `https://github.com/HRNet/HRNet-Semantic-Segmentation/blob/master/experiments/cityscapes/seg_hrnet_w18_small_v1_512x1024_sgd_lr1e-2_wd5e-4_bs_12_epoch484.yaml` verbatim the file used to produce 70.3% or does it need further hyperparameter tuning? (I'm on the `pytorch-v1.1` branch.)

In case it's helpful (although I'm sure this isn't informative), here are the cIoUs for the w18-v1 retrained model:

```
Loss: 0.179, MeanIU:  0.6509, Best_mIoU:  0.6521
[0.97245895 0.79921705 0.8969752  0.43651182 0.47062117 0.56336364
 0.57983322 0.68906234 0.91533262 0.60986547 0.93415257 0.74804671
 0.46804914 0.91671634 0.4241423  0.58802203 0.24108752 0.41514963
 0.69802723]
```

**Comment by sunke123** [comment]
Hi, sorry for the late reply.
This is my training log for seg_hrnet_w18_small_v1: https://1drv.ms/u/s!Aus8VCZ_C_33gSQ7irYs1DZy68yv?e=6fiuJN

The model 'hrnet_w18_for_mb' is the same as hrnet_w18_small_v1.
Please check it out.
I think that I use the same settings as you.
And If you use pytorch-v1.2, you can try to run this code on pytorch-v1.1. My friends tell me that they use pytorch-v1.2 and get worse  performance.

**Comment by alvinwan** [comment]
No problem, thanks for replying! I noticed that the config in your training log includes `CLASS_BALANCE: True` whereas the current YAML does not have this variable set (by default, `lib/config/default.py` sets this variable to false. I will try retraining with the class balance variable set to true, and if that works, I'll make a PR with the change.

**Comment by alvinwan** [comment]
For posterity, I was unable to reproduce `seg_hrnet_w18_small_v1`'s reported accuracy of `0.7026`. Training with pytorch 1.1 + `CLASS_BALANCE: True` did improve my initial accuracy above by 1.5%. However, I obtained `0.6688` and `0.6674` (~3.4% short)

@sunke123 would you happen to have the mean/std of your runs?

**Comment by sunke123** [comment]
I only ran it 1 once. 
I think the difference of 3.4% is too large.
Could you share your training log with me?
I will double-check it.

**Comment by alvinwan** [comment]
@sunke123 Thanks for offering to take a look. Looking at your logs again though, I noticed the first epoch results in ~30% mIOU (whereas my first epoch results in ~10% mIOU), and your logs contain an extra line:

```
2019-08-16 06:05:50,342 => loading pretrained model pretrained_models/hrnet_w18_for_mb.pth
```

Is that a mobilenet-v3 checkpoint pretrained on imagenet? If so, that likely explains the discrepancy (if so, oops, sorry). Just in case that's not the issue, here are both training logs:

- 0.6674 ([log](https://github.com/HRNet/HRNet-Semantic-Segmentation/files/3594249/seg_hrnet_w18_small_v1_512x1024_sgd_lr1e-2_wd5e-4_bs_12_epoch484_clsbal_2019-09-05-22-38_train.log))
- 0.6688 (log [part 1](https://github.com/HRNet/HRNet-Semantic-Segmentation/files/3594260/seg_hrnet_w18_small_v1_512x1024_sgd_lr1e-2_wd5e-4_bs_12_epoch484_clsbal_2019-09-07-19-41_train.log), [part 2](https://github.com/HRNet/HRNet-Semantic-Segmentation/files/3594264/seg_hrnet_w18_small_v1_512x1024_sgd_lr1e-2_wd5e-4_bs_12_epoch484_clsbal_2019-09-07-17-56_train.log))

**Comment by alvinwan** [comment]
@sunke123 sorry for bothering, that fixed it! haha. I appreciate your help. These are probably the first reproducible results I've ever seen. The code is clean, the results are reproducible... couldn't ask for more.

**Comment by sunke123** [comment]
Congrats! hh~
Thanks for your attention.
If you have any questions, please feel free to contact us.
